### PR TITLE
Vapi 815 Add Mute and Unmute to SDK

### PR DESCRIPTION
**Pull request by hkussie111**
****Description:**** 

This pull request introduces the ability to mute and unmute the audio stream within the Vapi class. The changes allow the application to handle mute-audio and unmute-audio commands received through the `onAppMessage` method. 

Additionally, this pull request addresses a TypeScript error encountered in the `vapi.ts` file, specifically within the event listeners for `"remote-participants-audio-level"` and `"app-message"`. 

**Key Changes:**

- Added handling for mute-audio and unmute-audio commands in the onAppMessage method.

- Created a new method handleMuteUnmuteCommands to process mute and unmute actions.

- Updated `VapiEventNames` and `VapiEventListeners` types to include audio-muted and audio-unmuted events.

- Added unit tests to ensure the mute and unmute functionality works as expected.

**How to Test:**

- Run `npm install` 

- Execute `npm test` to run the unit tests

### CODE REVIEW

File: vapi.ts [hunk 1]
```diff
@@ -200,14 +210,31 @@ export default class Vapi extends VapiEventEmitter {
     }
   }
 
+  private muteUnmuteLocalAudio(enable: boolean) {
```
Comment (jordancde) [inline vapi.ts:213]: preference for `setMuted(boolean)`

File: vapi.ts [hunk 2]
```diff
@@ -200,14 +210,31 @@ export default class Vapi extends VapiEventEmitter {
...
+      this.call.setLocalAudio(enable);
+    }
+  }
+
   private onAppMessage(e?: DailyEventObjectAppMessage) {
     if (!e) return;
     try {
       if (e.data === "listening") {
         return this.emit("call-start");
+      } else if (e.data == "mute-audio") {
```
Comment (jordancde) [inline vapi.ts:224]: Wait, I don't think mute-audio, unmute-audio, audio-unmuted are real events that come in through app-message. Unless I'm missing something.

I think app-message are exclusively json / text messages we send from our BE, not from Daily

File: vapi.ts [hunk 3]
```diff
@@ -178,9 +182,15 @@ export default class Vapi extends VapiEventEmitter {
...
+      
+      this.call.on("remote-participants-audio-level", (e) => {
+        if (e) this.handleRemoteParticipantsAudioLevel(e);
+      });
+
+      this.call.on("app-message", (e) => {
+        if (e) this.onAppMessage(e);
+      });
+
       this.call.on("app-message", (e) => this.onAppMessage(e));
```
Comment (jordancde) [inline vapi.ts:194]: Duplicate app-message handlers

File: __tests__/vapi.test.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    expect((vapi as any).muteUnmuteLocalAudio).toHaveBeenCalledWith(false);
+    expect(mockEmit).toHaveBeenCalledWith('audio-muted');
+  });
+
+  it('should unmute the audio and emit "audio-unmuted" on "unmute-audio" message', () => {
+    vapi['onAppMessage']({ data: 'unmute-audio' } as any); // Simulate receiving an unmute-audio message
+    expect((vapi as any).muteUnmuteLocalAudio).toHaveBeenCalledWith(true);
+    expect(mockEmit).toHaveBeenCalledWith('audio-unmuted');
+  });
+});
```
Comment (jordancde) [inline __tests__/vapi.test.ts:30]: Was this code tested physically? Or just mocked?

If not, time would be better spent spinning up an example and testing that all the events actually emit, etc.

Comment (jordancde) [reply __tests__/vapi.test.ts:30]: Was talking with nikhil: "I think this is the second test ever written on our platform" 🫢

File: vapi.ts (hunk 2 above)
Comment (hkussie111) [reply vapi.ts]: Yeah I dug through Daily’s documentation and it looks like the audio state is controlled exclusively through the DailyCall object

**Review by jordancde — APPROVED** [review]

**Pull request merged**